web/src/message_parser.ts

```diff
@@ -23,3 +23,7 @@ export function message_has_image(message: Message): boolean {
 export function message_has_attachment(message: Message): boolean {
     return is_element_in_message_content(message, "a[href^='/user_uploads']");
 }
+
+export function message_has_reaction(message: Message): boolean {
+    return message?.message_reactions?.length > 0;
```

Comment:
This seems wrong given the types. If you accept a `Message` object, we don't need to consider it being `undefined`. Likely you have a bug in the caller that you're working around.

Also the red symbol just below this comment indicates you deleted the newline at the end of this file; you should add one since that's always preferred, and maybe configure your editor to always generate them.